### PR TITLE
Weather changes

### PR DESCRIPTION
**Pull request by xslk**
## Description

Renamed overcast to cloudy. Updated color to reflect most current colors based on bulbapedia weather page.

weather

Not changing sunny to harsh sunlight because sunny is shorter and easier for everyone.

Updated weather boost and removed weather nerfs because damage nerfs are not fun so that we don't have to go with canon for that.

Changed weight of weathers to be uniform. 

Removed Hail and windy. 
Bulbapedia says that hail has been replaced by snow as of gen 9.
Strong wind only applies to one pokemon in ORAS while that pokemon is in battle, no overworld effects. 

## Motivation and Context

Changes weather weight so that all weathers have an equal chance to be experienced. Eliminate extra weather that

Provides all pokemon types a weather that boost their attack.

## How Has This Been Tested?

Tested on local and personal test site.

## Screenshots (optional):

Old weather Hoenn
old weather
New weather Hoenn
new weather
Old weather Galar
old galar
New weather Galar
new galar

## Types of changes

- New feature

### CODE REVIEW

**Comment by DraKay** [comment]
If you look at pogo is Canon don't have weather nerfs

**Comment by Farboo171** [comment]
linking to #3651

**Comment by Farboo171** [comment]
What were the weather debuff you decided not to include? Just sun and rain?

**Comment by xslk** [comment]
My initial draft for debuffs was 
Clear: ghost, dark
Cloudy: psychic, fairy
Rain: fire, rock
Thunderstorm: steel, flying
Snow: dragon, fighting
Blizzard: dragon, ground, fighting
Sunny: water, ice
Sandstorm: grass, electric 
Fog: bug, poison

Most were based on type weaknesses.

**Comment by Jaaslet** [comment]
I have spend some time trying to figure out what is behind all the changes, and i have no idea. Can you explain your choices for this PR?

**Comment by DraKay** [comment]
Afaik this pr was made because some types aren't boosted while other are boosted multiple times with current system and seem to make it more near to Canon as current system is a weird mix of games. (as op stated in first message windy doesn't really exist outside 1 hoenn game affecting only rayquaza in battle)

**Comment by Jaaslet** [comment]
> Afaik this pr was made because some types aren't boosted while other are boosted multiple times with current system and seem to make it more near to Canon as current system is a weird mix of games. (as op stated in first message windy doesn't really exist outside 1 hoenn game affecting only rayquaza in battle)

There are a lot more changes than that, in this PR.
Removing wind because it's only in one game is weird, when we can do so it only shows up in one game.

**Comment by xslk** [comment]
Hail is redundant, we don't need 3 ice boosting weathers in gen 8 and on hail is replaced by snow.

Windy only happens in the main line games with the ability delta stream, which only mega rayquaza has in ORAS while it is in battle, the effect on strong winds goes away as soon as its changed out of battle. Windy is also in POGO and pokemon mystery dungeon but if you go POGO or pokemon mystery dungeon as a reason why to keep windy then more extreme changes to weather would be needed.

Weather weights are universal since there is no more need to have any rare weather unless diamond dust is added later in time as an easter egg or universal booster during special events.

Debuffs were removed because no one really wants to do less damage unless they're trying to fail in the battle frontier. 

Damage buffs are all the same across the board. No type gets special treatment since weather weights are all universal. I don't feel that a over reliance to progress based on weather should be needed. I left the boost at 1.1x since I figured if it needed to be higher, discussions would be had behind the scenes by the dev team and and it would be changed by a dev or cc before being accepted.

File: src/modules/weather/Weather.ts [hunk 1]
```diff
@@ -32,37 +32,32 @@ export default class Weather {
...
+            new WeatherCondition(WeatherType.Cloudy, '#a7db8d', 'Clouds fill the skies.', 10,
+                [{ type: PokemonType.Fighting, multiplier: 1.1 }, { type: PokemonType.Poison, multiplier: 1.1 }, { type: PokemonType.Dark, multiplier: 1.1 }]),
         [WeatherType.Rain]:
             new WeatherCondition(WeatherType.Rain, '#9db7f5', 'It\'s rainy and humid.', 10,
-                [{ type: PokemonType.Water, multiplier: 1.1 }, { type: PokemonType.Bug, multiplier: 1.05 }]),
+                [{ type: PokemonType.Water, multiplier: 1.1 }, { type: PokemonType.Bug, multiplier: 1.1 }]),
         [WeatherType.Thunderstorm]:
-            new WeatherCondition(WeatherType.Thunderstorm, '#a19288', 'It\'s currently raining heavily with thunder.', 5,
-                [{ type: PokemonType.Electric, multiplier: 1.1 }, { type: PokemonType.Water, multiplier: 1.1 }, { type: PokemonType.Fire, multiplier: 0.9 }]),
+            new WeatherCondition(WeatherType.Thunderstorm, '#a19288', 'It\'s currently raining heavily with thunder.', 10,
```
Comment (Farboo171) [inline src/modules/weather/Weather.ts:44]:
```suggestion
            new WeatherCondition(WeatherType.Thunderstorm, '#a19288', 'It\'s currently raining heavily with thunder.', 5,
```

File: src/modules/weather/Weather.ts [hunk 2]
```diff
@@ -32,37 +32,32 @@ export default class Weather {
...
+                [{ type: PokemonType.Electric, multiplier: 1.1 }, { type: PokemonType.Water, multiplier: 1.10 }, { type: PokemonType.Dragon, multiplier: 1.1 }]),
```
Comment (Farboo171) [inline src/modules/weather/Weather.ts:45]:
```suggestion
                [{ type: PokemonType.Electric, multiplier: 1.1 }, { type: PokemonType.Water, multiplier: 1.1 }, { type: PokemonType.Dragon, multiplier: 1.1 }, { type: PokemonType.Fire, multiplier: 0.9 }]),
```

Comment (Farboo171) [reply src/modules/weather/Weather.ts:45]: Standardize "1.10" to "1.1", restores debuff for fire

File: src/modules/weather/Weather.ts [hunk 3]
```diff
@@ -32,37 +32,32 @@ export default class Weather {
...
-            new WeatherCondition(WeatherType.Snow, '#bbe6e6', 'It\'s cold and snowing.', 5,
-                [{ type: PokemonType.Ice, multiplier: 1.05 }]),
-        [WeatherType.Hail]:
-            new WeatherCondition(WeatherType.Hail, '#74e6e6', 'It\'s cold and hailing.', 3,
+            new WeatherCondition(WeatherType.Snow, '#bbe6e6', 'It\'s cold and snowing.', 10,
                 [{ type: PokemonType.Ice, multiplier: 1.1 }]),
         [WeatherType.Blizzard]:
-            new WeatherCondition(WeatherType.Blizzard, '#98d8d8', 'A howling blizzard blows.', 2,
-                [{ type: PokemonType.Ice, multiplier: 1.2 }, { type: PokemonType.Fire, multiplier: 0.9 }, { type: PokemonType.Grass, multiplier: 0.9 }]),
+            new WeatherCondition(WeatherType.Blizzard, '#98d8d8', 'A howling blizzard blows.', 10,
```
Comment (Farboo171) [inline src/modules/weather/Weather.ts:50]:
```suggestion
            new WeatherCondition(WeatherType.Blizzard, '#98d8d8', 'A howling blizzard blows.', 5,
```

Comment (Farboo171) [reply src/modules/weather/Weather.ts:50]: Lowers blizzard frequency, as there are two ice boosting weathers.

File: src/modules/weather/Weather.ts [hunk 4]
```diff
@@ -32,37 +32,32 @@ export default class Weather {
...
+                [{ type: PokemonType.Ice, multiplier: 1.1 }, { type: PokemonType.Steel, multiplier: 1.1 }]),
```
Comment (Farboo171) [inline src/modules/weather/Weather.ts:51]:
```suggestion
                [{ type: PokemonType.Ice, multiplier: 1.1 }, { type: PokemonType.Steel, multiplier: 1.1 }, { type: PokemonType.Grass, multiplier: 0.9 }]),
```

Comment (Farboo171) [reply src/modules/weather/Weather.ts:51]: Restores Grass debuff

File: src/modules/weather/Weather.ts [hunk 5]
```diff
@@ -32,37 +32,32 @@ export default class Weather {
...
         [WeatherType.Sunny]:
             new WeatherCondition(WeatherType.Sunny, '#f5ac78', 'The sunlight is strong.', 10,
-                [{ type: PokemonType.Fire, multiplier: 1.2 }, { type: PokemonType.Grass, multiplier: 1.1 }, { type: PokemonType.Water, multiplier: 0.9 }]),
+                [{ type: PokemonType.Fire, multiplier: 1.1 }, { type: PokemonType.Ground, multiplier: 1.1 }]),
```
Comment (Farboo171) [inline src/modules/weather/Weather.ts:54]:
```suggestion
                [{ type: PokemonType.Fire, multiplier: 1.1 }, { type: PokemonType.Ground, multiplier: 1.1 }, { type: PokemonType.Water, multiplier: 0.9 }]),
```

Comment (Farboo171) [reply src/modules/weather/Weather.ts:54]: Restores Water debuff

File: src/modules/weather/Weather.ts [hunk 6]
```diff
@@ -32,37 +32,32 @@ export default class Weather {
...
             new WeatherCondition(WeatherType.Sandstorm, '#d1c07d', 'A sandstorm is raging.', 10,
-                [{ type: PokemonType.Rock, multiplier: 1.1 }, { type: PokemonType.Ground, multiplier: 1.1 }, { type: PokemonType.Steel, multiplier: 1.1 }]),
+                [{ type: PokemonType.Rock, multiplier: 1.1 }, { type: PokemonType.Ground, multiplier: 1.1 }]),
         [WeatherType.Fog]:
             new WeatherCondition(WeatherType.Fog, '#d2c2ef', 'The fog is deep...', 10,
-                [{ type: PokemonType.Ghost, multiplier: 1.2 }, { type: PokemonType.Dark, multiplier: 1.1 }, { type: PokemonType.Electric, multiplier: 0.9 }]),
-        [WeatherType.Windy]:
-            new WeatherCondition(WeatherType.Windy, '#81c4ca', 'Mysterious strong winds blow.', 1,
-                [{ type: PokemonType.Flying, multiplier: 1.2 }, { type: PokemonType.Dragon, multiplier: 1.1 }]),
+                [{ type: PokemonType.Psychic, multiplier: 1.1 }, { type: PokemonType.Ghost, multiplier: 1.1 }, { type: PokemonType.Fairy, multiplier: 1.1 }]),
```
Comment (Farboo171) [inline src/modules/weather/Weather.ts:60]:
```suggestion
                [{ type: PokemonType.Psychic, multiplier: 1.1 }, { type: PokemonType.Ghost, multiplier: 1.1 }, { type: PokemonType.Fairy, multiplier: 1.1 }, { type: PokemonType.Electric, multiplier: 0.9 }]),
```

Comment (Farboo171) [reply src/modules/weather/Weather.ts:60]: Restores Electric debuff

**Review by Farboo171 — COMMENTED** [review]
These comments are the result of an internal CC discussion on this PR.

File: src/modules/weather/Weather.ts (hunk 6 above)
Comment (xslk) [reply src/modules/weather/Weather.ts:60]: If windy is being kept then a re-review of weather buffs would be needed as the original changes removes windy

**Pull request closed without merging**